Library/Homebrew/formulary.rb

```diff
@@ -949,6 +950,7 @@ def self.tap_loader_for(tapped_name, warn:)
       end
     end
 
+    name = name.split("/").last if name != :nil && type == :alias
```

Comment:
As mentioned in https://github.com/Homebrew/brew/pull/16623#discussion_r148542438, this is the exact same fix. Can you move the `split` into the conditional above to avoid checking another condition here?

Reply:
Yeah, sorry, your suggestion is simpler than what I proposed earlier. Using `String#split` should avoid the nil type errors entirely too.

Reply:
@apainintheneck I'm not sure who is targeted by @reitermarkus's wish to move the `split` - shall I do it? 
And if yes; I'm not sure where exactly to move it to?

So like this?
```diff
diff --git a/Library/Homebrew/formulary.rb b/Library/Homebrew/formulary.rb
index 9a1a7cc2c..cabc1f74a 100644
--- a/Library/Homebrew/formulary.rb
+++ b/Library/Homebrew/formulary.rb
@@ -909,7 +909,7 @@ module Formulary
     alias_name = tap.core_tap? ? name : "#{tap}/#{name}"
 
     if (possible_alias = tap.alias_table[alias_name].presence)
-      name = possible_alias
+      name = tap.core_tap? ? possible_alias : possible_alias.split("/").last
       type = :alias
     elsif (new_name = tap.formula_renames[name].presence)
       old_name = name
@@ -950,7 +950,6 @@ module Formulary
       end
     end
 
-    name = name.split("/").last if name != :nil && type == :alias
     path = find_formula_in_tap(name, tap)
     TapLoader.new(name, path, tap: tap)
   end
```

Reply:
Or like this?

```diff
diff --git a/Library/Homebrew/formulary.rb b/Library/Homebrew/formulary.rb
index 9a1a7cc2c..952ee50fd 100644
--- a/Library/Homebrew/formulary.rb
+++ b/Library/Homebrew/formulary.rb
@@ -948,9 +948,10 @@ module Formulary
       elsif Homebrew::API::Formula.all_formulae.key?(name)
         return FormulaAPILoader.new(name)
       end
+    elsif name != :nil && type == :alias
+      name = name.split("/").last
     end
 
-    name = name.split("/").last if name != :nil && type == :alias
     path = find_formula_in_tap(name, tap)
     TapLoader.new(name, path, tap: tap)
   end

```

Reply:
The latter seems to be error-prone … eg. when `tap.core_tap? && !Homebrew::EnvConfig.no_install_from_api?` is `true` and the API-lookup `Homebrew::API::Formula.all_formulae.key?(name)` is `false`.

Reply:
> @apainintheneck I'm not sure who is targeted by @reitermarkus's wish to move the `split` - shall I do it? And if yes; I'm not sure where exactly to move it to?
> 
> So like this?
> 
> ```diff
> diff --git a/Library/Homebrew/formulary.rb b/Library/Homebrew/formulary.rb
> index 9a1a7cc2c..cabc1f74a 100644
> --- a/Library/Homebrew/formulary.rb
> +++ b/Library/Homebrew/formulary.rb
> @@ -909,7 +909,7 @@ module Formulary
>      alias_name = tap.core_tap? ? name : "#{tap}/#{name}"
>  
>      if (possible_alias = tap.alias_table[alias_name].presence)
> -      name = possible_alias
> +      name = tap.core_tap? ? possible_alias : possible_alias.split("/").last
>        type = :alias
>      elsif (new_name = tap.formula_renames[name].presence)
>        old_name = name
> @@ -950,7 +950,6 @@ module Formulary
>        end
>      end
>  
> -    name = name.split("/").last if name != :nil && type == :alias
>      path = find_formula_in_tap(name, tap)
>      TapLoader.new(name, path, tap: tap)
>    end
> ```

This is pretty much what I was thinking. I'll update it now and test it locally to make sure it works.